pkg/vendir/fetch/cache/cache.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/vmware-tanzu/carvel-vendir/pkg/vendir/fetch/cache/resources"
+)
+
+// Cache functionality
+type Cache interface {
+	Hit(id string) (string, bool)
+	Save(id string, src string) error
+	CopyFrom(id string, dst string) error
+}
+
+// FolderCache cache storing the information into a folder in the OS
+type FolderCache struct {
+	folder  string
+	maxSize resources.Quantity
+}
+
+// NewCache creates a new cache
+// When cacheFolder is empty this constructor will provide a noop cache
+func NewCache(cacheFolder string, maxContentCacheableSize string) (Cache, error) {
+	if cacheFolder == "" {
+		return &NoCache{}, nil
+	}
+	q, err := resources.ParseQuantity(maxContentCacheableSize)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to process maximum amount allowed to cache: %s", err)
+	}
+	return &FolderCache{folder: cacheFolder, maxSize: q}, nil
+}
+
+// Hit checks if a particular entry in the cache is present
+// Returns the path to the entry and a flag information if the entry was found or not
+func (c FolderCache) Hit(id string) (string, bool) {
```

Comment:
nit: but to my mind "Hit" sounds more like a command than a query. of course "Present" has the same issues and is a longer word.  idk, for 3 letter words, ... "Has" ?

Reply:
I was trying to use the caching more common terms like hit/miss to provide describe the behavior, but I think this can be an easy change